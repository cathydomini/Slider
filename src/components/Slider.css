
.slider img{
    width:400px;
    height: 300px;
    object-fit:cover;
}

.slider div {
    position: relative;
    display: flex;
    justify-content: center;
    align-items:center;
    background-color:1 solid red;

}

.right-arrow{
    position:absolute;
    right:10%;
    top:50%;
    z-index: 10;
}

.left-arrow{
    position: absolute;
    left:10%;
    top:50%;
    z-index: 10;
}
