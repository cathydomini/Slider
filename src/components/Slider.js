 import React, { useState } from 'react';
 import './Slider.css'; 

 import {
  FaChevronRight,
  FaChevronLeft,
 } from "react-icons/fa";

 const Slider = ({ slides })=>{
   const [current , setCurrent] = useState(1);
   const len = slides.length;

   const nextSlide = ()=>{
        setCurrent(current == len -1 ? 0 : current + 1 );    
   }
   const prevSlide = ()=>{
    
      setCurrent(current === 0 ? len-1 : current -1 );
    
  }
    return (     
            <div className="slider"> 
                <FaChevronLeft
                className="left-arrow"
                onClick ={prevSlide}
              />
              <FaChevronRight
                className="right-arrow"
                onClick ={nextSlide}
              />
                {slides.map((slide, index) =>(
                  <div key={slide.id}>  
                   {index === current &&
                      (<img src= {slide.image} alt="" ></img>
                    )}
                  </div>  
                ))}              
            </div>
    )
}


export default Slider;