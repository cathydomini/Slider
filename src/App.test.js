import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Slider from './components/Slider';

Enzyme.configure({ adapter: new Adapter() })

const ImageData = [
  {
    id:1,
    image:
      "https://images.unsplash.com/photo-1546190255-451a91afc548?ixlib=rb-1.2.1",
  },
  {
    id:2,
    image:
      "https://images.unsplash.com/photo-1591348122449-02525d70379b?ixlib=rb-1.2.1",
  },
  {
    id:3,
    image:
      "https://images.unsplash.com/photo-1548802673-380ab8ebc7b7?ixlib=rb-1.2.1",
  },
  {
    id:4,
    image:
      "https://images.unsplash.com/photo-1577023311546-cdc07a8454d9?ixlib=rb-1.2.1",
  },
];

it('render slider component', () =>{  
  const wrapper = shallow(<Slider slides ={ImageData} />);
  expect(wrapper.find("src"));
});