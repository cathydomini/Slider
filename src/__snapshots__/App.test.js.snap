// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render slider component 1`] = `
<div>
  <img
    alt=""
    src=""
  >
    img
  </img>
</div>
`;
